addons/web/static/src/views/fields/section_note_list/section_note_list_field.js

```diff
@@ -0,0 +1,387 @@
+            } else {
+                return { ...col };
+            }
+        });
+    }
+
+    getSectionCellClass(column, record) {
+        const section = this.getHierarchyItem(record.id);
+        const cssClass = this.getCellClass(column, record);
+        return mergeClasses(cssClass, "fst-italic", {
+            "fw-bolder": section.isTopSection,
+        });
+    }
+
+    getSectionRowClass(record) {
+        const section = this.getHierarchyItem(record.id);
+        const cssClass = this.getRowClass(record);
+        return mergeClasses(cssClass, `o_section_row_level_${section.isTopSection ? 1 : 2}`);
+    }
+
+    getSectionColumns(record) {
+        const columns = this.getColumns(record);
+        const sectionColumns = columns.filter(
+            (col) =>
+                col.widget === "handle" ||
+                (col.type === "field" && col.name === this.props.sectionContentField)
+        );
+        return sectionColumns.map((col) => {
+            if (col.name === this.props.sectionContentField) {
+                return { ...col, colspan: columns.length - sectionColumns.length + 1 };
+            } else {
+                return { ...col };
+            }
+        });
+    }
+
+    isSortable() {
+        return false;
+    }
+
+    makeNoteCreationContext() {
+        return {
+            [`default_${this.props.displayTypeField}`]: DISPLAY_TYPES.NOTE,
+        };
+    }
+
+    makeSectionCreationContext(isSubSection) {
+        return {
+            [`default_${this.props.displayTypeField}`]: isSubSection
+                ? DISPLAY_TYPES.SUBSECTION
+                : DISPLAY_TYPES.SECTION,
+        };
+    }
+}
+
+export class SectionNoteListField extends X2ManyField {
+    static props = {
+        ...super.props,
+        displayTypeField: String,
+        noteContentField: String,
+        sectionContentField: String,
+    };
+    static components = {
+        ...super.components,
+        ListRenderer: SectionListRenderer,
+    };
+
+    get rendererProps() {
+        return {
+            ...super.rendererProps,
+            displayTypeField: this.props.displayTypeField,
+            isM2M: this.isMany2Many,
+            noteContentField: this.props.noteContentField,
+            sectionContentField: this.props.sectionContentField,
+        };
+    }
+}
+
+/** @type {import("registries").FieldsRegistryItemShape} */
+export const sectionNoteListField = {
+    ...x2ManyField,
+    component: SectionNoteListField,
+    extractProps(staticInfo, dynamicInfo) {
+        const props = x2ManyField.extractProps(staticInfo, dynamicInfo);
+        return {
+            ...props,
+            displayTypeField: staticInfo.options.display_type_field,
+            noteContentField: staticInfo.options.note_content_field,
+            sectionContentField: staticInfo.options.section_content_field,
+        };
+    },
+    relatedFields: (info) => {
+        const relatedFields = [];
+        if (info.options.displayTypeField) {
```

Comment:
I wouldn't add those fields automatically. This bypasses the onchange mechanism: as the fields aren't defined in the arch, the view processing in python, which generates the onchange spec, has a partial picture, so it may load to missing fields in the spec. It's the same problem we have with `fieldDependencies`. Another reason not to do it is that you enforce the field type (text/char). Basically, it could be the opposite and it could still work just fine.

Reply:
So I known it's a bit annoying, but I would just require those fields to be in the arch. At least, it would be 100% correct.